pkg/util/hostname/hostname_test.go

```diff
@@ -6,13 +6,11 @@
 //go:build !serverless
 // +build !serverless
 
-package util
+package hostname
```

Comment:
As mentioned when we spoke about this offline, if might make sense to extend the tested surface _before_ merging this in to make sure we do not change behavior inadvertently. Of course, we'd need some useful tests to implement, I believe our fallback logic might not be properly covered - maybe that's a good place to start.